build/sphinx/mongoc_common.py

```diff
@@ -1,12 +1,20 @@
 import os
+import re
+from pathlib import Path
+from typing import Any, Iterable, Sequence
+
 from docutils import nodes
-try:
-    from sphinx.util.compat import Directive
-except ImportError:
-    from docutils.parsers.rst import Directive
+from docutils.nodes import Node, document
+
+from sphinx.application import Sphinx
+from sphinx.application import logger as sphinx_log
+from sphinx.builders.dirhtml import DirectoryHTMLBuilder
+from sphinx.config import Config
+from sphinx.project import Project
+from sphinx.util.docutils import SphinxDirective
 
-needs_sphinx = '1.6'
-author = 'MongoDB, Inc'
+needs_sphinx = "1.6"
```

Comment:
```suggestion
needs_sphinx = "7.0"
```
Update Sphinx version (as done for Poetry)?

Reply:
Updated with 6.0, as that is the version available in Fedora Rawhide and used for the RPM build (for now)

Reply:
Updated again to 5.0, since we're building on Fedora 38 in CI (not Rawhide).